force-app/main/default/classes/TaxCalculationHandlerTest.cls

```diff
@@ -0,0 +1,97 @@
+@isTest
+private class TaxCalculationHandlerTest {
+    
+    // Utility method to create a Job Application record
+    private static Job_Application__c createJobApp(Decimal salary) {
+        Job_Application__c jobApp = new Job_Application__c();
+        jobApp.Name = 'Test Application';
+        jobApp.Salary__c = salary;
+        return jobApp;
+    }
+    
+    // Positive test to check all tax calculations and set values in each bracket
+    @isTest static void testTaxCalculationAcrossBrackets() {
+        List<Job_Application__c> jobApps = new List<Job_Application__c>{
+            createJobApp(10000),  
+            createJobApp(20000),  
+            createJobApp(50000),  
+            createJobApp(120000), 
+            createJobApp(250000)  
+        };
+        insert jobApps;
+        
+        // Test the tax calculation logic
+        Test.startTest();
+        TaxCalculationHandler.calculateTaxesAndTakeHomePay(jobApps);
+        Test.stopTest();
+        
+        // Check tax calculations and tax brackets
+        for (Job_Application__c jobApp : jobApps) {
+            Decimal expectedTax = TaxCalculationHandler.calculateFederalTax(jobApp.Salary__c);
+            System.assertEquals(expectedTax, jobApp.Federal_Income_Tax__c, 'Incorrect federal tax calculation for salary ' + jobApp.Salary__c);
+        }
+    }  
+    
+    // Positive test to verify getSalary and getTaxBracketDetails methods
+    @isTest static void testGetSalaryAndTaxDetails() {
+        Job_Application__c jobApp = createJobApp(50000);
+        insert jobApp;
+
+        Decimal fetchedSalary = TaxCalculationHandler.getSalary(jobApp.Id);
```

Comment:
Each method should be tested in isolation. Ideally, split these into two tests and be sure to wrap the method being tested inside the Test.startTest() and Test.stopTest() context